src/main/java/org/apache/sysml/api/ScriptExecutorUtils.java

```diff
@@ -81,6 +81,27 @@ public static void executeRuntimeProgram(Program rtprog, ExecutionContext ec, DM
 		DMLScript.SYNCHRONIZE_GPU = dmlconf.getBooleanValue(DMLConfig.SYNCHRONIZE_GPU);
 		DMLScript.EAGER_CUDA_FREE = dmlconf.getBooleanValue(DMLConfig.EAGER_CUDA_FREE);
 		DMLScript.STATISTICS_MAX_WRAP_LEN = dmlconf.getIntValue(DMLConfig.STATS_MAX_WRAP_LEN);
+		if(DMLScript.USE_ACCELERATOR) {
+			String prevDataType = DMLScript.DATA_TYPE; 
+			DMLScript.DATA_TYPE = dmlconf.getTextValue(DMLConfig.SYSML_DATA_TYPE);
+			if(!prevDataType.equalsIgnoreCase(DMLScript.DATA_TYPE)) {
+				if(DMLScript.DATA_TYPE.equalsIgnoreCase("double")) {
```

Comment:
Actually, on second thought, do you think its a good idea to abstract these out into its own class?
Something like `CudaDataTypeProperties`. Just a thought, leaving it here is ok too.
Since you have refactored out most of the other functions, maybe you could refactor out the aggregate functions.

Reply:
Done